pkg/spoke/statusfeedback/reader.go

```diff
@@ -0,0 +1,127 @@
+package statusfeedback
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/util/jsonpath"
+	workapiv1 "open-cluster-management.io/api/work/v1"
+	"open-cluster-management.io/work/pkg/spoke/statusfeedback/rules"
+)
+
+type StatusReader struct {
+	wellKnownStatus rules.WellKnownStatusRuleResolver
+}
+
+func NewStatusReader() *StatusReader {
+	return &StatusReader{
+		wellKnownStatus: rules.DefaultWellKnownStatusRule(),
+	}
+}
+
+func (s *StatusReader) GetValuesByRule(obj *unstructured.Unstructured, rule workapiv1.FeedbackRule) ([]workapiv1.FeedbackValue, error) {
+	errs := []error{}
+	values := []workapiv1.FeedbackValue{}
+
+	switch rule.Type {
+	case workapiv1.WellKnownStatusType:
+		paths := s.wellKnownStatus.GetPathsByKind(obj.GroupVersionKind())
+		if len(paths) == 0 {
+			return values, fmt.Errorf("cannot find the wellknown statuses for resrouce with gvk %s", obj.GroupVersionKind().String())
+		}
+
+		for _, path := range paths {
+			value, err := getValueByJsonPath(path.Name, path.Path, obj)
+			if err != nil {
+				errs = append(errs, err)
```

Comment:
you need to be sure that these errors include the `path.Name`, so you'll be able to debug.